data_model/src/asset.rs

```diff
@@ -411,18 +430,25 @@ impl DefinitionId {
     /// # Errors
     /// Fails if any sub-construction fails
     #[inline]
-    pub fn new(name: &str, domain_name: &str) -> Result<Self, ParseError> {
-        Ok(Self {
-            name: Name::new(name)?,
-            domain_id: DomainId::new(domain_name)?,
-        })
+    pub fn new(name: Name, domain_id: <Domain as Identifiable>::Id) -> Self {
```

Comment:
Also can be `const`

Reply:
Maybe we should enable [this](https://rust-lang.github.io/rust-clippy/rust-1.59.0/index.html#missing_const_for_fn) clippy lint?

Reply:
I think @appetrosyan experimented with this. I guess we haven't enabled it because it would involve too much changes?

Reply:
Feedback from @appetrosyan:
> Yes. I actually use it all the time. Problem is, it has lots of false-positives, so we can’t have it enabled in CI.

I think I can close this conversation now